tests/base_test.go

```diff
@@ -0,0 +1,649 @@
+		err = cl.Get(context.TODO(), localProviderIdentifier, returnedLocalProvider)
+
+		done = make(chan struct{})
+
+		// Wait for provider to be ready
+		statusCheck = func() {
+			err = cl.Get(context.TODO(), localProviderIdentifier, returnedLocalProvider)
+
+			if returnedLocalProvider.Status.Conditions.IsReady() {
+				log.Info("OCP Provider is ready")
+				close(done)
+			}
+		}
+
+		go wait.Until(statusCheck, time.Second, done)
+
+		select {
+		case <-timeout:
+			t.Errorf("Provider is not ready in time, last status: %v", returnedLocalProvider.Status)
+		case <-done:
+		}
+
+		ovirtCl := &Client{}
+		defer ovirtCl.Close()
+
+		err = ovirtCl.connect(secret, ovirtURL)
+		if err != nil {
+			t.Fatal(err, "Failed to connect ovirt sdk")
+		}
+		_, vmService, err := ovirtCl.getVMs(ref.Ref{ID: vmId})
+		if err != nil {
+			t.Fatal(err, "Failed to get vm")
+		}
+
+		diskAttachementResponse, err := vmService.DiskAttachmentsService().List().Send()
+		if err != nil {
+			t.Fatal(err, "Failed to get disk attachment service")
+		}
+
+		disks, ok := diskAttachementResponse.Attachments()
+		sdPairs = []v1beta1.StoragePair{}
+		if !ok {
+			t.Fatal("Failed to get disks")
+		}
+		for _, da := range disks.Slice() {
+			disk, ok := da.Disk()
+			if !ok {
+				t.Fatal("Failed to get disk")
+			}
+			diskService := ovirtCl.connection.SystemService().DisksService().DiskService(disk.MustId())
+			diskResponse, err := diskService.Get().Send()
+			if err != nil {
+				t.Fatal(err, "Failed to get disk")
+			}
+			sds, ok := diskResponse.MustDisk().StorageDomains()
+			if !ok {
+				t.Fatal("Failed to get sds")
+			}
+			for _, sd := range sds.Slice() {
+				sdId, ok := sd.Id()
+				if !ok {
+					t.Fatal("Failed to get sd id")
+				}
+				pair := v1beta1.StoragePair{
+					Source: ref.Ref{ID: sdId},
+					Destination: v1beta1.DestinationStorage{
+						StorageClass: sc,
+					},
+				}
+				sdPairs = append(sdPairs, pair)
+			}
+		}
+		nicsResponse, err := vmService.NicsService().List().Send()
+		nics, ok := nicsResponse.Nics()
+		if !ok {
+			t.Fatal("Failed to get nics")
+		}
+		nicPairs = []v1beta1.NetworkPair{}
+		for _, nic := range nics.Slice() {
+			vnicService := ovirtCl.connection.SystemService().VnicProfilesService().ProfileService(nic.MustVnicProfile().MustId())
+			vnicResponse, err := vnicService.Get().Send()
+			if err != nil {
+				t.Fatal(err, "Failed to get vnic service")
+			}
+			profile, ok := vnicResponse.Profile()
+			if !ok {
+				t.Fatal("Failed to get nic profile")
+			}
+
+			network, ok := profile.Network()
+			if !ok {
+				t.Fatal("Failed to get network")
+			}
+			networkId, ok := network.Id()
+			if !ok {
+				t.Fatal("Failed to get network id")
+			}
+			pair := v1beta1.NetworkPair{
+				Source: ref.Ref{ID: networkId},
+				Destination: v1beta1.DestinationNetwork{
+					Type: "pod",
+				},
+			}
+			nicPairs = append(nicPairs, pair)
```

Comment:
so we'll map all networks to the pod network? it doesn't seem to reflect a real world scenario..
but I see we also map all storages to the specified one, so we can keep it this way for now and change when we'll design the other cases to test